modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/plugin/descriptor/MulePluginModel.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.deployment.internal.plugin.descriptor;
+
+import static java.util.Optional.ofNullable;
+import org.mule.runtime.api.meta.model.ExtensionModel;
+
+import java.util.Optional;
+
+/**
+ * This object matches the mule-plugin.json element within a plugin. The describer holds information that has being
+ * picked up from the JSON file (and the pom.xml when implemented). There's no extra logic, such as calculating the
+ * URLs needed to feed a class loader or the {@link ExtensionModel} for the current plugin.
+ * <p/>
+ * The idea behind this object is to just load some bits of information that later each "loader" will consume to
+ * generate things like {@link ClassLoader}, {@link ExtensionModel}, etc.
+ *
+ * @since 4.0
+ */
+public class MulePluginModel {
```

Comment:
not to fix in this PR, but this class should be in mule-api.

The serializer for it, should be in a persistence sub-module of such api